diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +118,197 @@ class LineLength : Rule("line-length") {
         }
     }
 
+    private fun fixError(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        if (wrongNode.elementType == EOL_COMMENT) {
+            fixComment(wrongNode, configuration)
+            return
+        }
+        var parent = wrongNode
+        do {
+            when (parent.elementType) {
+                FUN -> {
+                    if (!parent.hasChildOfType(BLOCK))
+                        parent.addChild(PsiWhiteSpaceImpl("\n"), parent.findChildByType(EQ)!!.treeNext)
+                    return
+                }
+                CONDITION -> {
+                    fixCondition(parent, configuration)
+                    return
+                }
+                PROPERTY -> {
+                    fixProperty(parent, configuration)
+                    return
+                }
+                else -> parent = parent.treeParent
+            }
+        } while (parent.treeParent != null)
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration){
+        var node = wrongNode
+        do {
+            val commentText = node.text
+            var indexLastSpace = configuration.lineLength.toInt()
+            while (commentText[indexLastSpace] != ' ') {
+                --indexLastSpace
+                if (indexLastSpace == 0)
+                    return
+            }
+            val nodeText = "//${commentText.substring(indexLastSpace, commentText.length)}"
+            val newNode = LeafPsiElement(EOL_COMMENT, nodeText)
+            node.treeParent.addChild(LeafPsiElement(EOL_COMMENT, commentText.substring(0, indexLastSpace)), node)
+            node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node)
+            node.treeParent.addChild(newNode, node)
+            node.treeParent.removeChild(node)
```

Comment:
```suggestion
            node.treeParent.run {
                addChild(LeafPsiElement(EOL_COMMENT, commentText.substring(0, indexLastSpace)), node)
                addChild(PsiWhiteSpaceImpl("\n"), node)
                addChild(newNode, node)
                removeChild(node)
            }
```
This is just my personal opinion, by I find it really useful to wrap such operations on a specific object into `run` to avoid repetitive code and better separate a number of operations as a block. Feel free to leave as is, if you don't like this style.